test/integration/tags/inline_comment_test.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class InlineCommentTest < Minitest::Test
+  include Liquid
+
+  def test_tag_in_different_styles
+    assert_template_result('', '{% # This text gets ignored %}')
+    assert_template_result('', '{%# This text gets ignored #%}')
+    assert_template_result('', '{%# This text gets ignored %}')
+    assert_template_result('', '{%#- This text gets ignored -#%}')
+  end
+
+  def test_test_syntax_error
+    assert_template_result('fail', '{% #This doesnt work %}')
+
+    assert false 
+  rescue 
+    # ok good
+  end
+
+  def test_tag_ws_stripping
+    assert_template_result('', '   {%#- This text gets ignored -#%}    ')
```

Comment:
I don't understand the reason you are proposing supporting this style.  Does it have any semantic difference?

Other than this test case, it seems like `#` is supposed to treat the rest of the line as a comment, but here it seems to be integrated into the start tag.  So does this have different semantic for multi-line comments like the following?

```liquid
{%#- This is clearly a comment
  echo "but how about this line?" -#%}
```

I'm also unsure if you are proposing that the `-#%}` end tag should have any significance over `-%}`.